### PR TITLE
Testsuite enhancements for validating affinity processing

### PR DESCRIPTION
**Pull request by rachmatowicz**
This PR aims to add features to the EJB client testsuite in order to better validate affinity processing.

For details, see: https://issues.jboss.org/browse/EJBCLIENT-263

### CODE REVIEW

**Comment by rachmatowicz** [comment]
I'm posting this PR now just to show where its going. I'll be adding some test cases today to cover the issues mentioned in the associated JIRA.

**Comment by rachmatowicz** [comment]
Dont know how the commit for [EJBCLIENT-232] got associated with this PR ...

**Comment by rachmatowicz** [comment]
Ready to be merged.

**Comment by rachmatowicz** [comment]
Concerning the changes in EJBCLIENT-276, would it not be better to declare a subinterface of Echo called TransactionEcho or similar to hold the whoAreYou() and whoAreYouNonTx() methods, so we can keep that stuff out of the way of any tests which don't use it?  

With regard to returning both the echo result and the node on which it is executed, for purposes of  checking affinity, we already use a Result() return value in the Wildfly testsuite which supports result.getValue() and result.getNode(). Which is a little more obvious than whoAreYou().

**Comment by fl4via** [comment]
@rachmatowicz can you please rebase this and update it? I am assuming that this enhancement is still valid although a little bit old

**Comment by rachmatowicz** [comment]
Yes, I forgot about those changes ...
I managed to rebase successfully just now but need to look at what changes I made to see if they are still relevant.

**Comment by fl4via** [comment]
thanks @rachmatowicz !

File: src/test/java/org/jboss/ejb/client/test/TransactionTestCase.java [hunk 1]
```diff
@@ -107,24 +92,25 @@ public static void beforeClass() throws Exception {
...
-        server2.register(OTHER_APP, MODULE_NAME, DISTINCT_NAME, EchoBean.class.getSimpleName(), new EchoBean(SERVER2_NAME));
-        server3.register(OTHER_APP, MODULE_NAME, DISTINCT_NAME, EchoBean.class.getSimpleName(), new EchoBean(SERVER3_NAME));
-        server4.register(OTHER_APP, MODULE_NAME, DISTINCT_NAME, EchoBean.class.getSimpleName(), new EchoBean(SERVER4_NAME));
+        for (int i = 0; i < 4; i++) {
+            startServer(i, 6999 + (i*100), true);
+        }
+
+        deployStateless(0);
+        deployStateful(0);
+        // don't deploy on server 2
```
Comment (fl4via) [inline src/test/java/org/jboss/ejb/client/test/TransactionTestCase.java]: Maybe an extra information on server 2 being actually server index 1 would make this comment less confusing

File: src/test/java/org/jboss/ejb/client/test/TransactionTestCase.java [hunk 2]
```diff
@@ -107,24 +92,25 @@ public static void beforeClass() throws Exception {
...
+        deployStateless(2);
+        deployStateful(2);
+        deployStateless(3);
+        deployStateful(3);
+
+        // don't deploy on server 1
```
Comment (fl4via) [inline src/test/java/org/jboss/ejb/client/test/TransactionTestCase.java]: The same here, server 1 is server index 0

File: src/test/java/org/jboss/ejb/client/test/TransactionTestCase.java (hunk 1 above)
Comment (rachmatowicz) [reply src/test/java/org/jboss/ejb/client/test/TransactionTestCase.java]: Done.

File: src/test/java/org/jboss/ejb/client/test/TransactionTestCase.java (hunk 2 above)
Comment (rachmatowicz) [reply src/test/java/org/jboss/ejb/client/test/TransactionTestCase.java]: Done

**Comment by fl4via** [comment]
@rachmatowicz it appears that merging the other PR for affinity caused this branch to have conflicts, can you please look into it? Then I'll merge it

**Comment by rachmatowicz** [comment]
Hi @fl4via
I have fixed up the conflicts caused by the previous commit. All tests pass. Should be ready to go.

**Comment by tadamski** [comment]
@rachmatowicz could you please rebase this?

**Comment by rachmatowicz** [comment]
Closing this RP, it is superseded by https://github.com/wildfly/jboss-ejb-client/pull/378

**Pull request closed without merging**